Tasks/Common/nuget-task-common/CommandHelper.ts

```diff
@@ -121,11 +121,7 @@ export function isOnPremisesTfs(): boolean {
         return false;
     }
 
-    // not an ideal way to detect hosted, but there isn't a variable for it, and we can't make network calls from here
-    // due to proxy issues.
-    const collectionUri = tl.getVariable("System.TeamFoundationCollectionUri");
-    const parsedCollectionUri = url.parse(collectionUri);
-    return !(/\.visualstudio\.com$/i.test(parsedCollectionUri.hostname));
+    return tl.getVariable("System.ServerType") !== "Hosted";
```

Comment:
This must be a new predefined build variable, I don't see it documented. It might be safer to do a case-insensitive comparison.